lib/Sema/CSDiagnostics.cpp

```diff
@@ -2208,6 +2208,7 @@ bool MissingMemberFailure::diagnoseAsError() {
   }
 
   auto baseType = resolveType(getBaseType())->getWithoutSpecifierType();
+  auto baseExprType = getType(baseExpr)->getWithoutSpecifierType();
```

Comment:
I think this makes sense only in case of `isForKeyPathDynamicMemberLookup`. So I'd suggest following - let's add a `baseType` to `tryTypoCorrection` closure and have a separate block of code for handing keypath dynamic lookup cases, because it makes sense to emit custom diagnostic regardless of possibility to typo correct it.

Reply:
It should be placed right after original call to typo correction and look something like this:

```c++
if (getLocator()->isForKeyPathDynamicMemberLookup()) {
   auto baseExprType = ...;
   tryTypoCorrection(baseExprType);

   if (correction = corrections.claimUniqueCorrection()) {
      // `did you mean?` diagnostic
   } else {
      // tailored no such member diagnostic.
   }

   return true;
}
```